### PR TITLE
api: Add deploymentapi and subpackages from ecctl

### PR DESCRIPTION
**Pull request by marclop**
## Description

Moves the code from `github.com/elastic/ecctl/pkg/deployments` under the
`api` package and renames any subpackage which exposes an actual api to
include the suffix `api` to avoid potential confusions with other self-
contained packages.

Cleans up any of the previous `ecctl` references and moves the struct
`deploymentapi.ResourceParams` to `depresourceapi.Params`.

Also, cleans up the usages of `api.UnwrapError` for `apierror.Unwrap`
and moves the necessary helper packages from `ecctl` to various other
package locations (`util`, `mock`, etc).

## Related Issues

Resolves elastic/ecctl#275

## Motivation and Context

Doubling up the api in the `ecctl` client. Instead that API should live in the cloud-sdk-go.

## Types of Changes

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: pkg/api/deploymentapi/noteapi/add.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+// AddParams is consumed by Add.
+type AddParams struct {
+	Params
+	Message string
+	UserID  string
+}
+
+// Validate ensures the parameters are valid
+func (params AddParams) Validate() error {
+	var merr = multierror.NewPrefixed("note add")
```
Comment (karencfv) [inline pkg/api/deploymentapi/noteapi/add.go:39]:
```suggestion
	var merr = multierror.NewPrefixed("deployment note add")
```

File: pkg/api/deploymentapi/noteapi/show.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+// GetParams is used on List
+type GetParams struct {
+	Params
+	NoteID string
+}
+
+// Validate confirms the parmeters are valid
+func (params GetParams) Validate() error {
+	var merr = multierror.NewPrefixed("note get")
```
Comment (karencfv) [inline pkg/api/deploymentapi/noteapi/show.go:37]:
```suggestion
	var merr = multierror.NewPrefixed("deployment note get")
```

File: pkg/api/deploymentapi/noteapi/update.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+type UpdateParams struct {
+	Params
+	NoteID  string
+	UserID  string
+	Message string
+}
+
+// Validate confirms the parmeters are valid
+func (params UpdateParams) Validate() error {
+	var merr = multierror.NewPrefixed("note update")
```
Comment (karencfv) [inline pkg/api/deploymentapi/noteapi/update.go:40]:
```suggestion
	var merr = multierror.NewPrefixed("deployment note update")
```

File: pkg/api/deploymentapi/depresourceapi/elasticsearch_topology.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+// Sets ZoneCount to a DefaultDataZoneCount.
+func (element *ElasticsearchTopologyElement) fillDefaults() {
+	if element.ZoneCount == 0 {
+		element.ZoneCount = DefaultDataZoneCount
+	}
+}
+
+// Validate ensures the parameters are usable by the consuming function.
+func (element *ElasticsearchTopologyElement) Validate() error {
+	var merr = multierror.NewPrefixed("deployment topology")
```
[the remark this answers is not in the document]
Comment (marclop) [reply pkg/api/deploymentapi/depresourceapi/elasticsearch_topology.go:89]: Maybe better to just leave it as `elasticsearch topology`?

**Pull request merged**